modules/post/linux/gather/enum_containers.rb

```diff
@@ -0,0 +1,172 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Linux Container Enumeration',
+        'Description' => %q{
+          This module attempts to enumeratec containers on the target machine and optionally run a command on each active container found..
+          Supports Docker, LXC and RKT.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => ['stealthcopter'],
+        'Platform' => ['linux'],
+        'SessionTypes' => ['shell', 'meterpreter']
+      )
+    )
+    register_options(
+      [
+        OptString.new('CMD', [false, 'Optional command to run on each running container', ''])
+      ]
+    )
+  end
+
+  def cmd
+    datastore['CMD']
+  end
+
+  # Check if a container program is installed and usable by current user
+  def runnable(container_type)
+    case container_type
+    when 'docker'
+      command = 'docker container ls -a >/dev/null 2>&1 && echo true'
+    when 'lxc'
+      command = 'lxc list >/dev/null 2>&1 && echo true'
+    when 'rkt'
+      command = 'rkt list >/dev/null 2>&1 && echo true'
```

Comment:
Technically we shouldn't be doing this, we should just be running the commands on their own here. The reason for this is that listing the running containers often requires `root` privileges. Just running `rkt` or `docker` to see if the command is on the system aka the system **could** be running containers usually just requires local access as any user.

Reply:
*Update*: Interestingly when I installed `lxc`, it didn't require permissions to list running containers but `docker` and `rkt` did. Weird. Regardless we should be expecting that things might fail due to a lack of permissions and try minimize this possibility wherever possible.

Reply:
My reasoning for using the list commands here was that these are what are used later on to enumerate so it makes sense to use these commands as a check because if these fail we wont be able to list the containers. I think if we add in your previous suggestion of listing when the platforms are installed (using the `command_exists` function) even when 0 containers are found it will help with this.

Yeah the LXC listing containers is interesting. I **think** it's possible to enumerate all containers by browsing /proc without any special permissions so it makes sense why they've not restricted this by user. But I think researching / doing this would be beyond the scope of this module.

I have some time tomorrow and friday so will have a look at improving this, thanks again for the feedback!

Reply:
@stealthcopter I think I'm going to update this line a bit more as `rkt` wasn't playing nice with your `2>&1` part of the command, which made me think, since we are doing regex on the result, we don't really need all these extra parts of the command that could fail, since we just end up doing regex on the result returned from the command and then we never use the returned value ever again.

To this extent I am going to make you echo statement echo out `truthyTrue` instead, which will prevent it from matching on similar values in help menus, and which will allow us to just replace the `true` part of your existing regex with a value that is more likely not exist within any help menu (as at present its possible the output could be updated to legitimately have `true` at the end of a line, which will currently trigger the regex and return a false positive).

Reply:
@stealthcopter Sorry for the deleted last update, realized your code is actually needed to prevent some output from being shown to the screen. Error at the moment is that `rkt` doesn't play nice with the `2>&1` command because for some odd reason it decides that if `rkt` is not passed an argument that it will just go and error out. Seems a good solution to this is just to do `rkt help` instead which is accessible to all users and won't throw an error so going to try replacing that now.

Reply:
No problem, please make any changes you see fit and Ill test them tomorrow on all the platforms

Reply:
Fixed with last commit